### PR TITLE
Odroid h4 udpates

### PR DESCRIPTION
**Pull request by miczyg1**
Synced with upstream version a bit.

Also changed the power state after failure to match original behavior of the board.

After these changes the board can boot DTS from USB and Ubuntu from eMMC, which was not possible before.

### CODE REVIEW

File: src/mainboard/hardkernel/odroid_h4/devicetree.cb [hunk 1]
```diff
@@ -38,33 +38,29 @@ chip soc/intel/alderlake
...
-			register "usb3_ports[3]" = "USB3_PORT_EMPTY"
-			register "usb3_ports[4]" = "USB3_PORT_EMPTY"
-			register "usb3_ports[5]" = "USB3_PORT_EMPTY"
-			register "usb3_ports[6]" = "USB3_PORT_EMPTY"
-			register "usb3_ports[7]" = "USB3_PORT_EMPTY"
-			register "usb3_ports[8]" = "USB3_PORT_EMPTY"
-			register "usb3_ports[9]" = "USB3_PORT_EMPTY"
+			register "usb2_ports" = "{
+
+#define ODROID_H4_USB2_PORT_REAR { \
```
Comment (mkopec) [inline src/mainboard/hardkernel/odroid_h4/devicetree.cb:43]: does it not work with any of the already defined macros?

**Comment by pietrushnic** [comment]
@miczyg1 IIUC This would be the best version for HWIO training, correct?

**Comment by pietrushnic** [comment]
:thinking: this branch doesn't seem to have `./build.sh odroid_h4` support.

**Comment by pietrushnic** [comment]
I was testing this PR (to be precise, I tested [hwio_2024](https://github.com/Dasharo/coreboot/compare/dasharo...hwio_2024) branch), and it seemed that:
- the network does not work DTS 
- eMMC is not detected in UEFI Shell

**Comment by miczyg1** [comment]
> I was testing this PR (to be precise, I tested [hwio_2024](https://github.com/Dasharo/coreboot/compare/dasharo...hwio_2024) branch), and it seemed that:
> 
> * the network does not work DTS
> * eMMC is not detected in UEFI Shell

I also happened to notice that eMMC sometimes does not appear in UEIF Shell, but most of the times it does.

Regarding network boot, it always worked for me. Remember that you have to have your time and date to be set properly to boot over HTTPS...

> 🤔 this branch doesn't seem to have ./build.sh odroid_h4 support.

The script is added here: https://github.com/Dasharo/coreboot/pull/564 @macchian asked @PLangowski to fix the deployment, but it doesn't seem to be finished still...

**Pull request merged**